engine/apps/alerts/models/alert_group.py

```diff
@@ -1098,41 +1081,39 @@ def bulk_resolve(user: User, alert_groups: "QuerySet[AlertGroup]") -> None:
             send_alert_group_signal.apply_async((log_record.pk,))
 
     @staticmethod
-    def bulk_restart(user: User, alert_groups: "QuerySet[AlertGroup]") -> None:
-        AlertGroupLogRecord = apps.get_model("alerts", "AlertGroupLogRecord")
+    def bulk_resolve(user: User, alert_groups: "QuerySet[AlertGroup]") -> None:
+        # stop maintenance for maintenance incidents
+        alert_groups_to_stop_maintenance = alert_groups.filter(resolved=False, maintenance_uuid__isnull=False)
+        for alert_group in alert_groups_to_stop_maintenance:
+            alert_group.stop_maintenance(user)
 
-        root_alert_groups_unack = alert_groups.filter(
+        root_alert_groups_to_resolve = alert_groups.filter(
             resolved=False,
-            acknowledged=True,
             root_alert_group__isnull=True,
-            maintenance_uuid__isnull=True,  # don't restart maintenance incident
+            maintenance_uuid__isnull=True,
         )
-        dependent_alert_groups_unack = AlertGroup.all_objects.filter(root_alert_group__in=root_alert_groups_unack)
-        alert_groups_to_restart_unack = root_alert_groups_unack | dependent_alert_groups_unack
+        if root_alert_groups_to_resolve.count() == 0:
```

Comment:
probably `.exists` will be more efficient here
```suggestion
        if not root_alert_groups_to_resolve.exists():
```